sdk/core/core/internal/az_http_internal.h

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_HTTP_PIPELINE_INTERNAL_H
+#define _az_HTTP_PIPELINE_INTERNAL_H
+
+#include <az_http.h>
+#include <az_result.h>
+
+#include <_az_cfg_prefix.h>
+
+AZ_INLINE az_http_method az_http_method_get() { return AZ_SPAN_FROM_STR("GET"); }
```

Comment:
So now we need an az_http_transport.h file which will be included by Curl and other HTTP transport policy. These az_http_method-* functions go in this header file.